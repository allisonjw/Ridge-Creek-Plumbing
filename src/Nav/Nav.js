import React from 'react';
import { NavLink } from 'react-router-dom';
import './Nav.scss';
import { IoIosFlame } from 'react-icons/io';
import { AiFillTool } from 'react-icons/ai';

export const Nav = () => {
    return (
      <header>
          <div className="nav_banner" aria-label="banner for phone number and email">
            <a 
              href="tel:3604856605" 
              className="nav_banner-link"
              aria-label="phone number">
              360.485.6605
            </a>
            <a    
              href="mailto:garthm@ridgeplumbing.com" 
              target="_blank" 
              className="nav_banner-link"
              aria-label="email address"
              rel="noopener noreferrer">
              garthm@ridgeplumbing.com
            </a>
          </div>
        <div className="header_logo" aria-label="business logo">
        <img className="logo-img" alt="ridge creek plumbing logo" src={require('../images/logo-black.png')}></img>
          <h1>Ridgecreek</h1>
          <h2 className="nav_h2"><span className="nav_img"><AiFillTool/></span>Plumbing | Gas<span className="nav_img"><IoIosFlame/></span></h2>
        </div>
        <nav role="navigation" aria-label="main navigation">
          <ul className="all_navbar_links">
            <li className="navbar_item">
                <NavLink 
                activeClassName="navlink-is_active" 
                className="navlink_normal"
                to="/"
                >
                About
                </NavLink>
            </li>
            <li className="navbar_item">      
                <NavLink 
                activeClassName="navlink-is_active" 
                className="navlink_normal"
                to="/portfolio"
                >
                    Portfolio
                </NavLink>
            </li>
            <li className="navbar_item">       
                <NavLink  
                activeClassName="navlink-is_active" 
                className="navlink_normal"
                to="/services"
                >
                    Services
                </NavLink>
            </li>
            <li className="navbar_item">      
                <NavLink 
                activeClassName="navlink-is_active" 
                className="navlink_normal"
                to="/contact"
                >
                    Contact
                </NavLink>
            </li>   
        </ul>
      </nav>
      </header>
    )
};

export default Nav;