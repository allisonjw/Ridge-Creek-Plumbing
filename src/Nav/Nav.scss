@import '../SCSS/variables';
@import '../SCSS/media-queries';

header {
    -webkit-box-shadow: 0 0 10px 3px rgba(0,0,0,.3);
    box-shadow: 0 0 10px 3px rgba(0,0,0,.3);
    font-size: 1rem;
    -webkit-transition: all .8s linear;
    transition: all .8s linear;
    padding-bottom: 11%;
    @include lg {

    }
    @include md {

    }
    @include sm {
        padding-bottom: 0%;
        display: flex;
        flex-direction: column;
        align-items: center;
    }
}

.header_logo {
    float: left;
    padding: 3%;
    // width: 20%;
    h1 {
        font-size: 3.5em;
        color: #f48024;
        text-shadow: 0px 0px 5px #0f0600ab;
        // @include xl {
        //     font-size: 4.5em;
        // }
        @include lg {
            font-size: 3.8em;
        }
        @include sm {
            font-size: 3em;
        }
    }
    .nav_h2 {
        text-align: center;
        font-size: 1.8em;
        // @include xl {
        //     font-size: 2.2em;
        // }
        @include lg {
            font-size: 1.8em;
        }
        @include sm {
            font-size: 1.4em;
        }
    }
    .nav_img {
        margin-top: 5px;
    }
    // @include xl {
    //     padding: 1%;
    //     width: 26%;
    // }
    @include lg {
        padding: 1%;
        // width: 30%;
    }
}

.nav_banner {
    background-color: $main_color;
    height: 46px;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    .nav_banner-link {
        color: white;
        margin: 0 3%;
        font-weight: 700;
        text-decoration: none;
          &:hover {
              color: $accent_color;
          }
    }
    // @include xl {
    //     font-size: 1.4em;
    // }
    @include sm {
        width: 100%;
    }
}

nav {
    float: right;
    margin: 5%;
    @include lg {
        float: none;
    }
    @include md {

    }
    @include sm {

    }
    @include xs {

    }
}

.all_navbar_links {
    margin: 0;
    padding: 0;
    list-style: none;
    float: left;
    .navbar_item {
        cursor: pointer;
        float: left;
        margin: 0 0 0 32px;
        transition: all .2s ease;
        &:hover {
            transform: scale(1.1);
        }
    }
}

a.navlink_normal,
a.navlink-is_active {
    font-size: 1.5em;
    font-weight: 700;
    text-decoration: none;
    color: $main_color;
    &:hover {
        color: $accent_color;;
    }
    @include xl {
        font-size: 1.9em;
    }
}

a.navlink_normal {
    border: none;
}

a.navlink-is_active {
    border-bottom: 2px solid navy;
}