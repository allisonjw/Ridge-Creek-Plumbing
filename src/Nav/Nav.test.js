import React from 'react';
import { render } from '@testing-library/react';
import Nav from './Nav';
import { shallow } from 'enzyme';

describe('Nav', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Nav />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});