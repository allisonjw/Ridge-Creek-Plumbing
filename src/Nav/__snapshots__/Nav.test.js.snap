// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<header>
  <div
    aria-label="banner for phone number and email"
    className="nav_banner"
  >
    <a
      aria-label="phone number"
      className="nav_banner-link"
      href="tel:3604856605"
    >
      360.485.6605
    </a>
    <a
      aria-label="email address"
      className="nav_banner-link"
      href="mailto:Ridgecreekplumbingllc@gmail.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Ridgecreekplumbingllc@gmail.com
    </a>
  </div>
  <div
    aria-label="business logo"
    className="header_logo"
  >
    <h1>
      Ridgecreek
    </h1>
    <h2
      className="nav_h2"
    >
      <span
        className="nav_img"
      >
        <AiFillTool />
      </span>
      Plumbing | Gas
      <span
        className="nav_img"
      >
        <IoIosFlame />
      </span>
    </h2>
  </div>
  <nav
    aria-label="main navigation"
    role="navigation"
  >
    <ul
      className="all_navbar_links"
    >
      <li
        className="navbar_item"
      >
        <NavLink
          activeClassName="navlink-is_active"
          className="navlink_normal"
          to="/"
        >
          About
        </NavLink>
      </li>
      <li
        className="navbar_item"
      >
        <NavLink
          activeClassName="navlink-is_active"
          className="navlink_normal"
          to="/portfolio"
        >
          Portfolio
        </NavLink>
      </li>
      <li
        className="navbar_item"
      >
        <NavLink
          activeClassName="navlink-is_active"
          className="navlink_normal"
          to="/services"
        >
          Services
        </NavLink>
      </li>
      <li
        className="navbar_item"
      >
        <NavLink
          activeClassName="navlink-is_active"
          className="navlink_normal"
          to="/contact"
        >
          Contact
        </NavLink>
      </li>
    </ul>
  </nav>
</header>
`;
